sample/11-rxjava-2/episode.yaml

```diff
@@ -0,0 +1,24 @@
+number: 11
+title: "Episode 11: Migration to RxJava 2 with Artur Dryomov from Juno"
+date: "2017-05-15"
+duration: "00:52:46"
+peopleIds:
+  hosts:
+    - hannes.dorfmann
+  guests:
+    - artur.dryomov
+url: "https://github.com/artem-zinnatullin/TheContext-Podcast/blob/master/show_notes/Episode_11.md"
+discussionUrl: "https://podcast.online/episode/discussio://github.com/artem-zinnatullin/TheContext-Podcast/issues/71"
+file:
+  url: "https://artemzin.com/static/thecontext/episodes/The.Context.episode.11.mp3"
+  length: 50658006
```

Comment:
It is as easy as making a `HEAD` request.

```
$ curl -I "https://artemzin.com/static/thecontext/episodes/The.Context.episode.11.mp3" | grep "Content-Length"
Content-Length: 50658006
```

Reply:
I know, but not for `duration` ...

Reply:
We can store the file but it'll mess with our workflow a bit when say @sockeqwe creates a PR and shares edited file with us at the same time but we then provide feedback about file and he changes it multiple times while PR is open

Reply:
Actually I think it might work out pretty well.

* Episode PR includes audio and its metadata in form of YAML and Markdown changes.
* Audio should be updated accordingly.
* PR gets merged — feed and files are deployed automatically to the hosting.

This approach brings a couple of benefits.

* Automatic deployment.
* Kind of independence from GitHub releases since all necessary files are in the repository.
* Potentially it becomes possible to accept episodes from other people and it would be done automagically since a little of manual work is required in comparison with the current approach.

Reply:
yep, I'm not against that, just wanted to mention current workflow, doesn't mean it's good :)

Reply:
I think we (artem and I) discussed some time ago if we should include mp3 files in repo or not. Back then we decided to not do it because   this bloats git repo and increases costs for github and so on whereas adding mo3 to release section basically uoloada mp3 file to aws which is more efficient. I think cocoa pods had similar issues. Sure we are not a big player and hence we shouldn't care that much if it helps our workflow, it's just a reminder ...

Reply:
Auto deploy would be great! Not sure though how excatly this could work becaus on each pull request  everything is rebuild like whole feed.xml,  all shownotes / markdown filese). Should then all mp3 files uploaded again? maybe there are some workarounds like check with HTTP HEAD  (maybe etag). Looks like we need an incremental solution which is not the way CI usually work. Just something to keep in mind...

Reply:
Do we need to deploy from PRs though? We can deploy from master build

Reply:
Good idea, still same problem though (just less frequently)

Reply:
There is Git LFS. But [there is a 1 GB limitation](https://help.github.com/articles/what-is-my-disk-quota/#file-and-repository-size-limitations) 😢

Reply:
Yeah well, then we can't really store them in the repo :(